diff --git a/node_modules/@mswjs/interceptors/lib/IsomorphicRequest.d.ts b/node_modules/@mswjs/interceptors/lib/IsomorphicRequest.d.ts
index 0059aa3..8a4979a 100644
--- a/node_modules/@mswjs/interceptors/lib/IsomorphicRequest.d.ts
+++ b/node_modules/@mswjs/interceptors/lib/IsomorphicRequest.d.ts
@@ -5,7 +5,7 @@ export interface RequestInit {
     credentials?: RequestCredentials;
     body?: ArrayBuffer;
 }
-export declare class IsomorphicRequest {
+export declare class IsomorphicRequest<Body = any> {
     id: string;
     readonly url: URL;
     readonly method: string;
@@ -18,7 +18,7 @@ export declare class IsomorphicRequest {
     constructor(request: IsomorphicRequest);
     get bodyUsed(): boolean;
     text(): Promise<string>;
-    json<T = any>(): Promise<T>;
+    json(): Promise<Body>;
     arrayBuffer(): Promise<ArrayBuffer>;
     clone(): IsomorphicRequest;
 }
