diff --git a/node_modules/@vitejs/plugin-react/dist/index.js b/node_modules/@vitejs/plugin-react/dist/index.js
index 1fe5286..2428521 100644
--- a/node_modules/@vitejs/plugin-react/dist/index.js
+++ b/node_modules/@vitejs/plugin-react/dist/index.js
@@ -385,7 +385,8 @@ function viteReact(opts = {}) {
       if (/\.(mjs|[tj]sx?)$/.test(extension)) {
         const isJSX = extension.endsWith("x");
         const isNodeModules = id.includes("/node_modules/");
-        const isProjectFile = !isNodeModules && (id[0] === "\0" || id.startsWith(projectRoot + "/"));
+        // Switch to cwd so babel plugin will run outside of project root
+        const isProjectFile = !isNodeModules && (id[0] === "\0" || id.startsWith(process.cwd() + "/"));
         const plugins = isProjectFile ? [...babelOptions.plugins] : [];
         let useFastRefresh = false;
         if (!skipFastRefresh && !ssr && !isNodeModules) {
