diff --git a/node_modules/msw/lib/glossary-297d38ba.d.ts b/node_modules/msw/lib/glossary-297d38ba.d.ts
index d15724b..63421db 100644
--- a/node_modules/msw/lib/glossary-297d38ba.d.ts
+++ b/node_modules/msw/lib/glossary-297d38ba.d.ts
@@ -51,7 +51,7 @@ interface MockedRequestInit extends RequestInit$1 {
     referrerPolicy?: RequestReferrerPolicy;
     cookies?: Record<string, string>;
 }
-declare class MockedRequest<RequestBody extends DefaultBodyType = DefaultBodyType> extends IsomorphicRequest {
+declare class MockedRequest<RequestBody extends DefaultBodyType = DefaultBodyType> extends IsomorphicRequest<RequestBody> {
     readonly cache: RequestCache;
     readonly cookies: Record<string, string>;
     readonly destination: RequestDestination;
