diff --git a/node_modules/tunnel-rat/dist/index.js b/node_modules/tunnel-rat/dist/index.js
index a076198..f162cf6 100644
--- a/node_modules/tunnel-rat/dist/index.js
+++ b/node_modules/tunnel-rat/dist/index.js
@@ -1,4 +1,4 @@
-import { useLayoutEffect } from 'react';
+import { useEffect } from 'react';
 import create from 'zustand';
 
 function tunnel() {
@@ -11,7 +11,7 @@ function tunnel() {
       children
     }) => {
       const set = useStore(state => state.set);
-      useLayoutEffect(() => {
+      useEffect(() => {
         set({
           current: children
         });
